apps/backend/src/api/devices.py

```
@@ -153,3 +169,64 @@ async def get_device_summary_by_hostname(
+        raise HTTPException(status_code=500, detail=str(e)) from e
+    except SSHConnectionError as e:
+        raise HTTPException(status_code=503, detail=str(e)) from e
+    except Exception as e:
+        logger.error(f"Error getting drive health for {hostname}: {e}")
+        raise HTTPException(status_code=500, detail="Failed to get drive health.") from e
+
+
+@router.get("/{hostname}/logs")
+async def get_device_logs_by_hostname(
+    hostname: str = Path(..., description="Device hostname"),
+    service: Optional[str] = Query(None, description="Filter by service name"),
```

Ruff (0.12.2)

Use `X | None` for type annotations

Convert to `X | None`

(UP045)